src/gui/uithememanager.cpp

```diff
@@ -32,13 +32,71 @@
 #include <QApplication>
 #include <QFile>
 #include <QIcon>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QPalette>
 #include <QResource>
 
 #include "base/iconprovider.h"
 #include "base/logger.h"
 #include "base/preferences.h"
 #include "base/utils/fs.h"
 
+namespace
+{
+    template<int N>
```

Comment:
Why do you really need it to be template? What kind of optimization you imagine in it?

Reply:
Needed fo pass an array and then use for range loop

Reply:
Why do you need this over optimization? It's not so critical case. This code will be executed only once. And there is no much elements to care.

I would drop this trivial function at all. Then declare some appropriate struct:
```c++
struct ColorDescriptor
{
    QString id;
    QPalette::ColorGroup colorGroup;
    QPalette::ColorRole colorRole;
};
```
Wrap your colors in it and put into array (or QVector) and apply theme all inside single loop.

I hope I didn't miss anything important...